src/book/03-guides/02-tools/03-webots.mdx

```diff
@@ -76,51 +76,68 @@ sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen
 
 ## Install Webots
 
-1. Open a terminal and swap to your home directory (note that Webots should be cloned in your home directory). Then clone the repository and change into the cloned directory:
+1. Open a terminal, clone the NUWebots repository and change into the cloned directory:
 
    ```bash
    cd ~
-   git clone --branch nubots-changes --single-branch --recurse-submodules -j$(nproc) https://github.com/NUbots/webots.git
-   cd webots
+   git clone --recurse-submodules -j$(nproc) https://github.com/NUbots/NUWebots.git
+   cd NUWebots
    ```
 
-   By using these commands, you'll be on the correct branch (`nubots-changes`) and you'll have the required submodules.
+   The NUWebots repo contains both NUbots-specific Webots files and the files needed to install Webots itself.
+   The first command can be changed to any directory you would like. If you're not sure, just follow what is in this guide.
+
+2. This step assumes you have done the previous step and are now in the NUWebots directory. Add the `WEBOTS_HOME` env variable to your `.bashrc` file:
+
+   ```
+   echo export WEBOTS_HOME=$(pwd)/webots >> ~/.bashrc
+   source ~/.bashrc
+   ```
 
-2. Install Webots build dependencies:
+3. Move into the webots directory and install the Webots build dependencies:
 
    ```bash
+   cd webots
    sudo ./scripts/install/linux_compilation_dependencies.sh
    ```
 
-3. Build Webots:
+4. Build Webots:
 
    ```bash
    make -j$(nproc)
    ```
 
-4. Add the `WEBOTS_HOME` env variable to your `.bashrc` file:
+## Running the Official RoboCup Simulation in Webots
 
+### Set Up the Protos
+
+1. Set up the symlink for the NUgus proto:
+
+   ```bash
+   ln -s $WEBOTS_HOME/../protos/robot/NUgus $WEBOTS_HOME/projects/samples/contests/robocup/protos
```

Comment:
```suggestion
   cp -r $WEBOTS_HOME/../protos/robot/NUgus $WEBOTS_HOME/projects/samples/contests/robocup/protos
```
The link does not work, copying does. I am not sure if there is a better way to do this because it feels odd to have two files in different locations

Reply:
Upon research it looks like `ln -r` may work, will test later today

Reply:
It is possible that `ln` may work with no flags, will test later today

Reply:
^^ It does not, I tried a few other flags too with not much luck

Reply:
Copying isn't what we want because then everyone will need to recopy whenever the model is updated. 
We definitely want the symbolic link. What happens when you run the command? An error or just nothing happens? I assume it's not the path since you said copying works.

Reply:
It looks like webots doesn't count a symlink as a real directory when it's looking through the `protos` directory subtree.
Textrures works as a symlink as does `NUgus.proto` and `postures.json` but the entire `NUgus_mesh` directory seems to need to be a normal directory, not a symlink.

![image](https://media.discordapp.net/attachments/407317825519353894/872940360307978270/unknown.png)

Reply:
This was the closest I could get without errors, most of them can be symlinks just not `NUgus_mesh`

```mkdir $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus
ln -s $WEBOTS_HOME/../protos/robot/NUgus/NUgus_textures $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus
cp -r $WEBOTS_HOME/../protos/robot/NUgus/NUgus_mesh $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus 
ln $WEBOTS_HOME/../protos/robot/NUgus/postures.json $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus 
ln -s $WEBOTS_HOME/../protos/robot/NUgus/NUgus.proto $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus```

Reply:
The other option is to have a giant list of each individual protos and make a symlink for each of those, because the files can be symlinks but not the directory that houses them. Is there a way to do like a for loop to do that neatly? For each file in a directory make a symlink to another directory?

Reply:
Not as pretty but this works, with symlinks. The `NUgus_mesh` directory is still a regular directory but the contents are symlinks which should update alongside the ones in the main `NUWebots` directory

```bash
mkdir $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus && mkdir $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus/NUgus_mesh
ln -s $WEBOTS_HOME/../protos/robot/NUgus/NUgus_textures $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus
ln -s $WEBOTS_HOME/../protos/robot/NUgus/postures.json $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus 
ln -s $WEBOTS_HOME/../protos/robot/NUgus/NUgus.proto $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus
for i in $(ls $WEBOTS_HOME/../protos/robot/NUgus/NUgus_mesh); do ln -s $WEBOTS_HOME/../protos/robot/NUgus/NUgus_mesh/$i $WEBOTS_HOME/projects/samples/contests/robocup/protos/NUgus/NUgus_mesh; done

Reply:
I'm concerned because if we change the files inside the NUgus folder then we'll have to get everyone to redo their symlinks.

Reply:
I absolutely stumbled on the solution.
`src/webots/vrml/WbProtoList.cpp`, line 73. Remove `QDir::NoSymLinks` from the flags expression being passed to `dir.entryInfoList`.

Here's a diff
```diff
diff --git a/webots/src/webots/vrml/WbProtoList.cpp b/webots/src/webots/vrml/WbProtoList.cpp
index d15da180..785abccf 100644
--- a/webots/src/webots/vrml/WbProtoList.cpp
+++ b/webots/src/webots/vrml/WbProtoList.cpp
@@ -69,7 +69,7 @@ void WbProtoList::findProtosRecursively(const QString &dirPath, QFileInfoList &p
     protoList.append(dir.entryInfoList(filter, QDir::Files, QDir::Name));
   }
   // search in subfolders
-  QFileInfoList subfolderInfoList = dir.entryInfoList(QDir::AllDirs | QDir::NoSymLinks | QDir::NoDotAndDotDot);
+  QFileInfoList subfolderInfoList = dir.entryInfoList(QDir::AllDirs | QDir::NoDotAndDotDot);
 
   if (!inProtos) {
     // try to identify a project root folder
```

Reply:
@ysims The man has done it. Works like a charm

Reply:
Looks like it's all good now! :tada: